subsys/net/ip/connection.c

```diff
@@ -757,6 +797,10 @@ enum net_verdict net_conn_input(struct net_pkt *pkt,
 				if (proto != ETH_P_ALL && proto != IPPROTO_RAW) {
 					continue; /* wrong protocol */
 				}
+			} else if (IS_ENABLED(CONFIG_NET_SOCKETS_INET_RAW) && raw_ip_pkt) {
+				if (conn->proto != 0) {
```

Comment:
Should we do the check like this
```
if (conn->proto != IPPROTO_IP) {
```
to make it clear what we are checking against (value 0 is a bit vague).